src/Akka.Persistence.Sql/Journal/Dao/BaseByteArrayJournalDao.cs

```diff
@@ -436,10 +439,11 @@ protected static ImmutableList<Exception> BuildWriteRejections(List<Util.Try<Jou
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static IQueryable<long> MaxMarkedForDeletionMaxPersistenceIdQuery(
             AkkaDataConnection connection,
-            string persistenceId)
+            string persistenceId, 
+            long maxSequenceNr)
             => connection
                 .GetTable<JournalRow>()
-                .Where(r => r.PersistenceId == persistenceId && r.Deleted)
+                .Where(r => r.PersistenceId == persistenceId && r.SequenceNumber <= maxSequenceNr)
```

Comment:
Optimization, instead of checking against persistence id and deleted field, check against persistence id and sequence number instead, as they are indexed.